tikv-client/src/main/java/com/pingcap/tikv/meta/TiDAGRequest.java

```diff
@@ -534,19 +556,17 @@ public TiTableInfo getTableInfo() {
 
   public TiDAGRequest setTableInfo(TiTableInfo tableInfo) {
     this.tableInfo = requireNonNull(tableInfo, "tableInfo is null");
+    setPhysicalId(tableInfo.getId());
     return this;
   }
 
-  public List<Long> getIds() {
```

Comment:
This might have performance issue when a lot of partitions exist in table.

getIds() will be called for each partition's rdd.